src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -77,6 +77,16 @@ def getModuleInfoForEnvironment(env)
     end
   end
 
+  def getAllModuleInfoForEnvironment()
+    all_envs = @env_loader.list
+    all_envs.collect do |env|
```

Comment:
I wonder if you did it as a single hash if it might make the code in Clojure which walks back over the map a little simpler?